common/http/class.Request.php

```diff
@@ -237,15 +242,31 @@ public function send()
         //curl_setopt($curlHandler, CURLINFO_HEADER_OUT, 1);
         //curl_setopt($curlHandler, CURLOPT_HEADER, true);
 
+        //directly setting `FOLLOWLOCATION` to false to make sure next lines would be working as expected
+        //and we can get the redirect url from curl
+        curl_setopt($curlHandler, CURLOPT_FOLLOWLOCATION, 0);
```

Comment:
Also, a silly question.
Cannot we just do something like this instead of all that's added below?
```suggestion
        curl_setopt($curlHandler, CURLOPT_FOLLOWLOCATION, (int)$followRedirects);
```

Reply:
Totally valid question.
Though, unfortunately, due to usage of `Authorization` header. If I correctly understood, from cURL 7.58.0 - it doesn't do so: https://curl.se/docs/CVE-2018-1000007.html

Reply:
OK. One last suggestion.

WDYT about adding checks on the `$redirectUrl` before trying it?
Like, if it's still pointing to the same domain+path. I assume only the request schema should change from http to https.